scipy/optimize/_minimize.py

```diff
@@ -585,6 +587,23 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
             if isinstance(bounds, Bounds):
                 bounds = new_bounds_to_old(bounds.lb, bounds.ub, x0.shape[0])
 
+    all_constraint_types = (NonlinearConstraint, LinearConstraint, dict)
```

Comment:
This should go to a separate helper function, maybe called
`normalize_constraints_to_new_format`?

Reply:
Also, the `bounds` would need to be normalized.

Reply:
@pv You've written:
"Also, the bounds would need to be normalized."
and in the original issue:
"Automatic conversion of the "old-style" ... bounds specifications to the "new" format"
I think the code ending two lines above your comment already addresses this. The new solver accepts old-style bounds, and the old solvers accept new-style bounds. The tests I wrote use old-style bounds with the new solver. I've found that `None`s are not converted to `np.inf` as they should be, and there aren't enough tests (and I can fix those things), but what do you mean when you write that the bounds needs to be normalized/converted? It seems this was already done.

I did want to put this all in a separate function, but instead I followed the style of the bounds conversion code and left it inline. I suppose I'll move both to separate functions in _constraints.py.

If I'm going to do this, I'd prefer to take all of the logic out and just write functions:
`bounds = standardize_bounds(bounds, method)`
`constraints = standardize_constraints(constraints, method)`
which makes the bounds/constraints appropriate for the method.

Reply:
Regarding conversion in the other direction, you mentioned in the original issue that an error message would also be acceptable "if [conversion is] not possible". I took "if that's not possible" to mean if it is not implemented. It's not implemented, so I raised an error. I would prefer to leave that for a separate PR because I did not originally intend to implement it and it will be a bit more involved. This PR is for making "trust-constr accept constraint dict".